springql-core/src/api/spring_config.rs

```diff
@@ -74,70 +13,51 @@ pub struct SpringConfig {
-    pub fn new(overwrite_config_toml: &str) -> Result<Self> {
-        let default_conf = config::Config::builder()
-            .add_source(config::File::from_str(
-                SPRING_CONFIG_DEFAULT,
-                config::FileFormat::Toml,
-            ))
-            .build()
-            .expect("SPRING_CONFIG_DEFAULT is in wrong format");
-
-        let c = config::Config::builder()
-            .add_source(default_conf)
-            .add_source(config::File::from_str(
-                overwrite_config_toml,
-                config::FileFormat::Toml,
-            ))
-            .build()
-            .map_err(|e| SpringError::InvalidFormat {
-                s: overwrite_config_toml.to_string(),
-                source: e.into(),
-            })?;
-
-        c.try_deserialize()
-            .map_err(|e| SpringError::InvalidConfig { source: e.into() })
-    }
-
-    /// Configuration by TOML format string.
-    ///
-    /// # Parameters
-    ///
-    /// - `overwrite_config_toml`: TOML format configuration to overwrite default. See `SPRING_CONFIG_DEFAULT` in [spring_config.rs](https://github.com/SpringQL/SpringQL/tree/main/springql-core/src/api/spring_config.rs) for full-set default configuration.
-    ///
-    /// # Failures
-    ///
-    /// - [SpringError::InvalidConfig](crate::api::error::SpringError::InvalidConfig) when:
-    ///   - `overwrite_config_toml` includes invalid key and/or value.
-    /// - [SpringError::InvalidFormat](crate::api::error::SpringError::InvalidFormat) when:
-    ///   - `overwrite_config_toml` is not valid as TOML.
-    pub fn from_toml(overwrite_config_toml: &str) -> Result<SpringConfig> {
-        SpringConfig::new(overwrite_config_toml)
+        Self {
+            worker: SpringWorkerConfig {
+                n_generic_worker_threads: 1,
+                n_source_worker_threads: 1,
+            },
+            memory: SpringMemoryConfig {
+                upper_limit_bytes: 10_000_000,
+                moderate_to_severe_percent: 60,
+                severe_to_critical_percent: 95,
+                critical_to_severe_percent: 80,
+                severe_to_moderate_percent: 40,
+                memory_state_transition_interval_msec: 10,
+                performance_metrics_summary_report_interval_msec: 10,
+            },
+            web_console: SpringWebConsoleConfig {
+                enable_report_post: false,
+                report_interval_msec: 3000,
+                host: "127.0.0.1".to_string(),
+                port: 8050,
+                timeout_msec: 3000,
+            },
+            source_reader: SpringSourceReaderConfig {
+                net_connect_timeout_msec: 1000,
+                net_read_timeout_msec: 100,
+                can_read_timeout_msec: 100,
+            },
+            sink_writer: SpringSinkWriterConfig {
+                net_connect_timeout_msec: 1000,
+                net_write_timeout_msec: 100,
+            },
+        }
```

Comment:
Don't want to hold default value both in https://github.com/SpringQL/SpringQL/pull/216/files#diff-56de30082a9c2e6459497fd78a8e2110be421f0674fa1c106ec2dbedfa952a71R12 and here.

How about changing `springql-core:: ... ::SpringConfig` into `spring-core:: ... ::Config`, which does not impl `Default`?

Then only `springql::SpringConfig` impl `Default` using `SPRING_CONFIG_DEFAULT`.

Reply:
this Default implementation used by only tests.

I move into `test_support`

https://github.com/SpringQL/SpringQL/pull/216/files#diff-7af0b986cf0abbb68962c12dbea4a8d1cd69bc2be040c92fe58c7a412848c61a

Reply:
No. It's used in online documents and some users (at least me).

https://springql.github.io/get-started/install-and-run-app#create-a-bin-crate-to-write-a-demo
<img width="895" alt="image" src="https://user-images.githubusercontent.com/498788/176772499-97299ce1-ddeb-4b1d-af75-f38f6a930ef6.png">

Reply:
Ah, sorry. I might mistakenly get what you say.

I realized that:

- Default values in TOML is (maybe) a good idea.
- TOML will be moved to springql crate.
- springql-core needs some way to provide Default impl for tests inside the crate.

Umm...